index.d.ts

```diff
@@ -0,0 +1,11 @@
+declare namespace CBOR {
+  export function encode(data: any, format?: "buffer"): Buffer;
+  export function encode(data: any, format: "hex"|"base64"): string;
```

Comment:
And `hex` can return a `Buffer`:

https://github.com/ARMmbed/cbor-sync/blob/master/main.js#L592

Reply:
No, it cannot. The only construction of `HexWriter` is:
https://github.com/ARMmbed/cbor-sync/blob/0b3dcad537433f4099562dd549681abdb988aa39/main.js#L611-L613

It passes `finalFormat=undefined` to `HexWriter` constructor, and therefore `this.finalFormat` is "hex".
https://github.com/ARMmbed/cbor-sync/blob/0b3dcad537433f4099562dd549681abdb988aa39/main.js#L565

Consequently, `Buffer.from` would not be executed.
https://github.com/ARMmbed/cbor-sync/blob/0b3dcad537433f4099562dd549681abdb988aa39/main.js#L591-L593